pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/Node.java

```diff
@@ -21,6 +26,9 @@
  */
 public interface Node {
 
+    // COMMENT: is it ok to take the opportunity on PMD 7 to rename this API and take out of there the methods
+    // that are only needed for javaCC implementations?
```

Comment:
yes, we can move away JavaCC specific methods and have them defined in `AbstractNode`. However, doing so requires we deprecate the methods here and define them in `AbstractNode` as part of a 6.x release first. This can be done, but requires a separate PR.

In particular, `jjtOpen`, `jttClose`, `jjtSetParent` are very specific to how JavaCC creates the AST. `jjtGetParent` is actually a basic operation for transversal, so you will need it (even if the name isn't very neutral with the `jjt` prefix)

Reply:
Nice, we thought that would be the case! 😄 
We will send a PR on 6.X then, would it be wise to deprecate all non-javaCC methods in orther to change their names, or the trouble is not worth it?

Reply:
deprecate non-javaCC methods? I think you meant to deprecate javaCC methods to rename them… we could, it wasn't in our agenda, but we could